components/consent-mgt/org.wso2.carbon.identity.consent.mgt/src/main/java/org/wso2/carbon/identity/consent/mgt/listener/ConsentDeletionAppMgtListener.java

```diff
@@ -135,6 +136,50 @@ public boolean doPostDeleteApplication(String applicationName, String tenantDoma
         return true;
     }
 
+    /**
+     * When an application is disabled, it will delete all relevant receipts issued against that application.
+     *
+     * @param serviceProvider Service provider which is disabled.
+     * @param tenantDomain    Tenant domain of the application.
+     * @param userName        Username of the person who does the deletion.
+     * @return true.
+     * @throws IdentityApplicationManagementException IdentityApplicationManagementException.
+     */
+    @Override
+    public boolean doPostUpdateApplication(ServiceProvider serviceProvider, String tenantDomain, String userName)
```

Comment:
seems like there is lot of common things in this method and `doPostDeleteApplication` lets extract this into a common method and use in both places.

Reply:
lets write a unit test for this

Reply:
https://github.com/wso2/carbon-identity-framework/pull/5732/commits/4f0851d547ebf25fe2eb6f4867950a1153a67222